utils.ts

```diff
@@ -38,6 +46,76 @@ export const makeQueryPeliasCompatible = (queryString: string): string => {
   return queryString.replace(/@/g, ' ').replace(/&/g, ' ')
 }
 
+export const hereResultTypeToPeliasLayer = (resultType: string): string => {
+  switch (resultType) {
+    case 'place':
+      return 'venue'
+    case 'houseNumber':
+      return 'address'
+    default:
+      return resultType
+  }
+}
+
+/**
+ * Executes a geocoder request using HERE API via @otp-ui/geocoder
+ * @param service Enum speicfying the type of API request to make.
+ * @param queryStringParam Query string from AWS with the url parameters from client.
+ * @param apiKey HERE API Key
+ * @returns HERE response in GeoJSON format.
+ */
+export const fetchHere = async (
+  service: string,
+  queryStringParams: Record<string, string>,
+  apiKey?: string
+): Promise<FeatureCollection> => {
+  const hereGeocoder = getGeocoder({
+    apiKey,
+    type: 'HERE'
+  })
+
+  const params = new URLSearchParams(queryStringParams)
+  const hereParams: AutocompleteQuery & SearchQuery & ReverseQuery = {}
+
+  // convert QSP into API call for hereGeocoder
```

Comment:
What is QSP? I'm guessing QueryStringParamater but we can probably make that a bit clearer